components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -120,35 +122,33 @@ class AtmosphereInput
   virtual ~AtmosphereInput () = default;
 
   // --- Methods --- //
+  // In case the class was constructed with the minimal ctor, these methods
+  // allow to finalize initialization later.
+  // NOTE: these two init methods are mutually exclusive
+  void init (const std::shared_ptr<const fm_type>& field_mgr,
+             const std::shared_ptr<const gm_type>& grids_mgr = nullptr);
+  void init (const std::shared_ptr<const grid_type>& grid,
+             const std::map<std::string,view_1d_host>& host_views_1d,
+             const std::map<std::string,FieldLayout>&  layouts);
 
   // Read fields that were required via parameter list.
   void read_variables (const int time_index = -1);
   int read_int_scalar (const std::string& name);
   void finalize();
 
 protected:
-  // Internal functions
-  // Sets up the scorpio metadata to preare for reading
-  // Inputs:
-  //  - field_mgr: the FieldManager containing the Field's where the
-  //               variables from the input filed will be read into.
-  //               Fields can be padded/strided.
-  void init();
 
-  // Sets up the scorpio metadata to preare for reading
-  // Inputs:
-  //  - host_views_1d: the 1d flattened views where data will be read into.
-  //                   These views must be contiguous (no padding/striding).
-  //  - layouts: the layout of the vars (used to reshape the views).
-  void set_parameters (const ekat::ParameterList& params);
+  void set_parameters (const ekat::ParameterList& params, const std::string& grid_name = "");
```

Comment:
What does it mean for the grid name to be left blank here?

Reply:
This is a poor design choice on my end. It all stems from how the parameter list is shaped, and allowing "simple" syntax for unit tests.

The general use case requires the fields names to be specified like so:
```
Fields:
  Grid Name 1:
     Fields Names: [a, b, c, ...]
  Grid Name 2:
     Fields Names: [a2, b2, ...]
```
But for single-grid tests, it seemed silly to request the user to know/set the grid name, so I allowed the syntax
```
Fields Names: [a, b, c, ...]
```
directly in the root parameter list. Now, the class offers a minimal ctor, where only comm and params are passed. This can be used if a) you don't want to read fields, only some global attributes (e.g., some counters), or if b) you need to read in some info first, before being able to pass in the grids (e.g., spa first reads ncols from nc file, then creates the views, then calls `init`, passing in the newly created views).

So the ctor where the field mgr (and the grid) is passed can call `set_parameters` passing the grid name, and therefore read the sublist `Fields->GridName->Field Names`. OTOH, the other ctors will not pass a grid name, so `set_parameters` will try to look for the `Field Names` entry, and if not found leave `m_fields_names` unset.

Reply:
NOTE: I'm not fond of how this ended up. An alternative, is to store the parameter list internally, and query it later for the fields names (if/when the grid name is already available).